src/Widgets/Views/MainView.vala

```diff
@@ -109,19 +113,38 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
     }
 
     private void update_devices_box_visible () {
-        devices_box.no_show_all = (devices_box.get_children ().length () <= 1);
-        devices_box.visible = !devices_box.no_show_all;
+        bool has_visible_device = false;
+        foreach (var device in devices.values) {
+            if (!device.no_show_all) {
```

Comment:
Maybe we could check for `visible` here instead of `no_show_all`? I think the visible property is more clear to understand here rather than a `no_show_all` property that's controlling not it's visibility but whether it's shown in containers using `show_all ()`.